vantage/budget_resource_model.go

```diff
@@ -0,0 +1,237 @@
+package vantage
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	resource_budget "github.com/vantage-sh/terraform-provider-vantage/vantage/resource_budget"
+	modelsv2 "github.com/vantage-sh/vantage-go/vantagev2/models"
+)
+
+// let the budget, budget performance and budget period models
+// defined in the resource_budget package be the common models for
+// both the data source and the resource
+type budgetModel resource_budget.BudgetModel
+type budgetPerformanceModel resource_budget.PerformanceValue
+type budgetPeriodResourceModel struct {
```

Comment:
i hate that we need separate ways to describe the period depending on whether it is for a data source or a resource, but cant find a better workaround for this.